### PR TITLE
Fix allowing to upload media without checking terms and condition check box

### PR DESCRIPTION
**Pull request by malavasita**

### CODE REVIEW

File: app/assets/js/rtMedia.js [hunk 1]
```diff
@@ -207,7 +207,7 @@ jQuery( 'document' ).ready( function( $ ) {
 
         // When Ajax completed attach media uploader to new activity, applay popup and attach media to comment uploader.
         jQuery(document).ajaxComplete( function(event, xhr, settings) {
-            if ( bp_template_pack && bp_template_pack === 'nouveau' ) {  
+            if ( bp_template_pack && bp_template_pack === 'nouveau' ) {
```
Comment (SID177) [inline app/assets/js/rtMedia.js]: Use yoda check here @malavasita

File: app/assets/js/rtMedia.js [hunk 2]
```diff
@@ -207,7 +207,7 @@ jQuery( 'document' ).ready( function( $ ) {
...
                 var get_action = get_parameter( 'action', settings.data );
                 if ( ( 'activity_filter' === get_action ||  'post_update' === get_action || 'get_single_activity_content' === get_action || 'activity_get_older_updates' === get_action ) && typeof rtmedia_masonry_layout != 'undefined' && rtmedia_masonry_layout == 'true' && typeof rtmedia_masonry_layout_activity != 'undefined' && rtmedia_masonry_layout_activity == 'true' ) {
```
Comment (SID177) [inline app/assets/js/rtMedia.js]: Use yoda here also @malavasita

File: app/assets/js/rtMedia.js [hunk 3]
```diff
@@ -1241,3 +1241,31 @@ function get_parameter( parameter, data ) {
...
+		var aw_whats_new_submit = jQuery('#aw-whats-new-submit');
+		// Check if I agree terms exist or not in activity page
+		if (rtmedia_upload_terms_conditions.length > 0) {
+			// Bydefault prevent user to post update.
+			aw_whats_new_submit.attr('disabled', 'disabled');
+			// Handle check event of I agree term.
+			rtmedia_upload_terms_conditions.change(function () {
+				if (rtmedia_upload_terms_conditions.prop('checked')) {
+					aw_whats_new_submit.removeAttr('disabled');
+				}
```
Comment (SID177) [inline app/assets/js/rtMedia.js]: JFI: If checkbox is checked then you remove attribute `disabled`. What if checkbox is first checked and then unchecked, have you tested the scenarios properly? @malavasita

Comment (malavasita) [reply app/assets/js/rtMedia.js]: Yes I have checked with possible scenarios. @SID177

File: app/assets/js/rtMedia.js [hunk 4]
```diff
@@ -206,10 +206,10 @@ function rtmedia_init_action_dropdown( parent ) {
 jQuery( 'document' ).ready( function( $ ) {
 
         // When Ajax completed attach media uploader to new activity, applay popup and attach media to comment uploader.
-        jQuery(document).ajaxComplete( function(event, xhr, settings) {
-            if ( bp_template_pack && bp_template_pack === 'nouveau' ) {  
+        jQuery( document ).ajaxComplete( function(event, xhr, settings) {
```
Comment (milindmore22) [inline app/assets/js/rtMedia.js]: alignment

File: app/assets/js/rtMedia.js [hunk 5]
```diff
@@ -1241,3 +1241,31 @@ function get_parameter( parameter, data ) {
...
+ * Check if upload term is active then force user to check I agree term.
+ */
+function rtm_allow_upload_terms_activity() {
+
+	jQuery('#whats-new').click(function () {
+		var rtmedia_upload_terms_conditions = jQuery('#rtmedia_upload_terms_conditions');
+		var aw_whats_new_submit = jQuery('#aw-whats-new-submit');
+		// Check if I agree terms exist or not in activity page
+		if (rtmedia_upload_terms_conditions.length > 0) {
+			// Bydefault prevent user to post update.
```
Comment (milindmore22) [inline app/assets/js/rtMedia.js]: By Default?

File: languages/buddypress-media.po [hunk 6]
```diff
@@ -4,7 +4,11 @@ msgid ""
 msgstr ""
 "Project-Id-Version: rtMedia for WordPress, BuddyPress and bbPress 4.4.8\n"
 "Report-Msgid-Bugs-To: http://community.rtcamp.com/c/rtmedia/\n"
+<<<<<<< HEAD
```
Comment (milindmore22) [inline languages/buddypress-media.po]: please resolve merge conflicts

**Comment by malavasita** [comment]
@milindmore22 Changes done :+1:

File: app/assets/js/rtMedia.js [hunk 7]
```diff
@@ -206,18 +206,18 @@ function rtmedia_init_action_dropdown( parent ) {
...
-                        rtmedia_activity_masonry();
-                        rtmedia_activity_stream_comment_media();
-                    } , 1000 );
-                }
-            }
-        } );
+		jQuery( document ).ajaxComplete( function( event, xhr, settings ) {
+			if ( bp_template_pack && 'legacy' !== bp_template_pack ) {
+				var get_action = get_parameter( 'action', settings.data );
+				if ( ('activity_filter' === get_action || 'post_update' === get_action || 'get_single_activity_content' === get_action || 'activity_get_older_updates' === get_action ) && 'undefined' !== typeof rtmedia_masonry_layout && 'true' === rtmedia_masonry_layout && 'undefined' !== typeof rtmedia_masonry_layout_activity && 'true' === rtmedia_masonry_layout_activity ) {
```
Comment (mangesh) [inline app/assets/js/rtMedia.js]: @malavasita Have you tested this `if` condition?
Make sure `rtmedia_masonry_layout` and `rtmedia_masonry_layout_activity` contains string value `'true'` and not boolean `true`.

Comment (malavasita) [reply app/assets/js/rtMedia.js]: Yes. It is coming from `rtMedia` options. and over there we are storing `string` only. @mangesh

**Comment by milindmore22** [comment]
I am seeing only po file changes where are js files change gone 🤔

**Comment by malavasita** [comment]
@milindmore22 Closing this PR coz it was having issues.
Please check [1126](https://github.com/rtMediaWP/rtMedia/pull/1128)

**Pull request closed without merging**